src/index.js

```diff
@@ -0,0 +1,225 @@
+      const errMsg = `datastore key does not have a valid format`
+
+      log.error(errMsg)
+      return callback(errcode(new Error(errMsg), 'ERR_INVALID_DATASTORE_KEY'))
+    }
+
+    if (!Buffer.isBuffer(val)) {
+      const errMsg = `received value is not a buffer`
+
+      log.error(errMsg)
+      return callback(errcode(new Error(errMsg), 'ERR_INVALID_VALUE_RECEIVED'))
+    }
+
+    const stringifiedTopic = key._buf.toString()
+
+    log(`publish value for topic ${stringifiedTopic}`)
+
+    // Publish record to pubsub
+    this._pubsub.publish(stringifiedTopic, val, callback)
+  }
+
+  /**
+   * Try to subscribe a topic with Pubsub and returns the local value if available.
+   * @param {Key} key identifier of the value to be subscribed.
+   * @param {function(Error, Buffer)} callback
+   * @returns {void}
+   */
+  get (key, callback) {
+    if (!(key instanceof Key)) {
+      const errMsg = `datastore key does not have a valid format`
+
+      log.error(errMsg)
+      return callback(errcode(new Error(errMsg), 'ERR_INVALID_DATASTORE_KEY'))
+    }
+
+    const stringifiedTopic = key._buf.toString()
+
+    log(`subscribe values for key ${stringifiedTopic}`)
+
+    // Subscribe
+    this._pubsub.subscribe(stringifiedTopic, this._handleSubscription, (err) => {
+      if (err) {
+        const errMsg = `cannot subscribe topic ${stringifiedTopic}`
+
+        log.error(errMsg)
+        return callback(errcode(new Error(errMsg), 'ERR_SUBSCRIBING_TOPIC'))
+      }
+
+      this._getLocal(key, callback)
+    })
+  }
+
+  // Get record from local datastore
+  _getLocal (key, callback) {
+    this._datastore.get(key, (err, dsVal) => {
+      if (err) {
+        const errMsg = `local record requested was not found for ${key._buf.toString()}`
+
+        log.error(errMsg)
+        return callback(errcode(new Error(errMsg), 'ERR_NO_LOCAL_RECORD_FOUND'))
+      }
+
+      if (!Buffer.isBuffer(dsVal)) {
+        const errMsg = `found record that we couldn't convert to a value`
+
+        log.error(errMsg)
+        return callback(errcode(new Error(errMsg), 'ERR_INVALID_RECORD_RECEIVED'))
+      }
+
+      callback(null, dsVal)
+    })
+  }
+
+  // handles pubsub subscription messages
+  _handleSubscription (msg) {
+    const { data, from, topicIDs } = msg
+    const key = topicIDs[0]
+
+    log(`message received for ${key} topic`)
+
+    // Stop if the message is from the peer (it already stored it while publishing to pubsub)
+    if (from === this._peerId.toB58String()) {
+      log(`message discarded as it is from the same peer`)
+      return
+    }
+
+    // verify if the received record is better
+    this._isBetter(key, data, (err, res) => {
+      if (!err && res) {
+        this._storeRecord(key, data)
+      }
+    })
+  }
+
+  // Validate record according to the received validation function
+  _validateRecord (value, peerId, callback) {
+    if (typeof this._validator !== 'object' || typeof this._validator.validate !== 'function') {
```

Comment:
Is it optional to pass a validator to the datastore? If not, we should assert in the constructor, otherwise this is fine.